geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/AbstractRedisData.java

```diff
@@ -200,6 +202,15 @@ public void fromDelta(DataInput in) throws IOException, InvalidDeltaException {
         byte[] byteArray = DataSerializer.readByteArray(in);
         applyDelta(new AppendDeltaInfo(byteArray, sequence));
         break;
+      case ZADDS:
+        int numMembers = DataSerializer.readPrimitiveInt(in);
```

Comment:
If you make the changes I suggested to ZAddsDeltaInfo this could would look like this:
int numMembers = DataSerializer.readPrimitiveInt(in);
ZAddsDeltaInfo delta = new ZAddsDeltaInfo(numMembers);
for (int i = 0; i < numMembers; i++) {
  byte[] member = DataSerializer.readByteArray(in);
  double score = DataSerializer.readPrimitiveDouble(in);
  delta.add(member, score);
}
applyDelta(delta);

Reply:
done